### ISSUE TITLE
False-positive E1101: Method '<no-name>' has no 'connect' member (no-member)

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#10650]
### Bug description

```python
from PySide6.QtWidgets import QApplication, QPushButton

app = QApplication()
button = QPushButton()
button.clicked.connect(quit)
button.show()
app.exec()
```

### Configuration

```ini
[tool.pylint]
# https://stackoverflow.com/q/56726580
extension-pkg-allow-list = ["PySide6"]
```

### Command used

```shell
pylint bug.py
```

### Pylint output

```python
************* Module bug
bug.py:1:0: C0114: Missing module docstring (missing-module-docstring)
bug.py:5:0: E1101: Method '<no-name>' has no 'connect' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

No E1101

### Pylint version

```shell
pylint 4.0.0
astroid 4.0.1
Python 3.13.8 (main, Oct  7 2025, 15:31:04) [MSC v.1944 64 bit (AMD64)]
```

### OS / Environment

Windows 11

### Additional dependencies

```python
pyside6-essentials==6.10.0
```

**Comment by jorisvr** [comment]
I have the same issue, both under Windows 11 and Debian 13.
Basically every Qt signal/slot connection is falsely flagged.

I tested the Python file reported above with a few different versions of Pylint and Astroid.
- The issue occurs with Astroid 4.0.0a0 and Pylint 4.0.0.
- The issue does not occur with Astroid 3.3.11 and Pylint 3.3.9.
- The issue occurs with Astroid 4.0.0a0 and Pylint 3.3.9.
- This suggests to me that the issue is caused by Astroid, not Pylint itself.

The issue seems related to pylint-dev/pylint-qt#1

**Comment by Gergzilla** [comment]
I also have this same issue but with PyQt6.  Even attempting to put the other modules in the whitelist do not affect it 

Pylint output:
```__main__.py:11:0: E0401: Unable to import 'PyQt6.QtCharts' (import-error)
__main__.py:11:0: E0611: No name 'QtCharts' in module 'PyQt6' (no-name-in-module)
__main__.py:13:0: E0401: Unable to import 'PyQt6.QtGui' (import-error)
__main__.py:13:0: E0611: No name 'QtGui' in module 'PyQt6' (no-name-in-module)
__main__.py:14:0: E0401: Unable to import 'PyQt6.QtWidgets' (import-error)
__main__.py:14:0: E0611: No name 'QtWidgets' in module 'PyQt6' (no-name-in-module)
```

Variations of the whitelist to try and resolve:
```
extension-pkg-whitelist=PyQt6, PyQt6.QtGui, PyQt6.QtCharts, PyQt6.QtWidgets
extension-pkg-whitelist=PyQt6, PyQt6-Qt6, PyQt6-sip
extension-pkg-whitelist=PyQt6, QtGui, QtCharts, QtWidgets
```
I understand that the error is not within my code but the multitude of errors in the linting is frustrating.  I havent tried enabling all c-extension code but I doubt that will fix it given the nature of the error.
This is on:
- pylint-4.0.5-py3-none-any.whl.metadata

(sorry for the edits, formatting issues)

**Comment by emmanuel-ferdman** [comment]
@bersbersbers @Gergzilla @jorisvr `pylint-qt` plugin was just released and is available on [PyPI](https://pypi.org/project/pylint-qt/)! 🎉
It fixes those Qt false-positives. Please give it a try:
```sh
pip install pylint-qt
pylint --load-plugins=pylint_qt --extension-pkg-allow-list=PySide6 your_code.py
```
Or in `pyproject.toml`:
```
[tool.pylint.main]
load-plugins = ["pylint_qt"]
extension-pkg-allow-list = ["PySide6"]
```
repo: https://github.com/pylint-dev/pylint-qt

**Comment by bersbersbers** [comment]
@emmanuel-ferdman thanks, looks promising. I'll have to wait for https://github.com/pylint-dev/pylint-qt/issues/2 and https://github.com/pylint-dev/pylint-qt/issues/3, though. Best of luck fixing those!
